qos/noop/context.go

```diff
@@ -28,18 +32,22 @@ type requestContext struct {
 	receivedResponses []endpointResponse
 
 	// presetFailureResponse, if set, is used to return a preconstructed response to the user.
-	// This is used by the consutrctor of the requestContext instance, e.g. if reading the HTTP request's body fails.
+	// This is used by the conductor of the requestContext instance, e.g. if reading the HTTP request's body fails.
 	presetFailureResponse *HTTPResponse
 }
 
 // GetServicePayload returns the payload to be sent to a service endpoint.
 // Implements the gateway.RequestQoSContext interface.
 func (rc *requestContext) GetServicePayload() protocol.Payload {
-	return protocol.Payload{
+	payload := protocol.Payload{
 		Method:          http.MethodPost,
```

Comment:
This will only support `POST` method, is that the intention for the first iteration?
If so, a TODO would be really helpful to future readers.

Reply:
Definitely great catch!